### ISSUE TITLE
Errors executing support diagnostic 6.1

### ISSUE THREAD

**Issue by inqueue** [elastic/elasticsearch-support-diagnostics#178]
Support diagnostics is producing execution errors.

Command: `./diagnostics.sh -h json5x60-4 -s -u elastic -p --noVerify`
```
DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=16.04
DISTRIB_CODENAME=xenial
DISTRIB_DESCRIPTION="Ubuntu 16.04.2 LTS"
```

Elasticsearch 6.0.0-rc1 DEB installation. 

Output files are created, but the tarball is not. Errors:

```
2017-10-10 18:58:36,070 ERROR - Error processing log and config files.
java.io.FileNotFoundException: Source '/usr/share/elasticsearch/config' does not exist
...
2017-10-10 18:58:36,071 ERROR - Error redacting config passwords.
java.lang.NullPointerException: null
        at com.elastic.support.diagnostics.commands.ScrubPasswordsCmd.execute(ScrubPasswordsCmd.java:37) ~[support-diagnostics-6.1.jar:6.1]
...
2017-10-10 18:58:36,073 ERROR - Error encountered running diagnostic. See logs for additional information.  Exiting application.
java.lang.RuntimeException: java.lang.NullPointerException
        at com.elastic.support.diagnostics.commands.ScrubPasswordsCmd.execute(ScrubPasswordsCmd.java:49) ~[support-diagnostics-6.1.jar:6.1]
...
2017-10-10 18:58:36,073 ERROR - Execution Error
java.lang.RuntimeException: Diagnostic runtime error
        at com.elastic.support.diagnostics.chain.DiagnosticChainExec.runDiagnostic(DiagnosticChainExec.java:44) ~[support-diagnostics-6.1.jar:6.1]
```
See details in [diagnostics.log.gz](https://github.com/elastic/elasticsearch-support-diagnostics/files/1373122/diagnostics.log.gz).

**Comment by inqueue** [comment]
Forgot to mention, the diagnostic was executed on and against a coordinating node.

**Comment by nemonster** [comment]
@inqueue This was addressed in the last round of fixes, although there were a couple of things going on here.

First, for most package installs:
./diagnostics.sh -h json5x60-4 -s -u elastic -p --noVerify

would cause an issue if you're not logged in as root or running with sudo. Remotes should be OK, but most user accounts can't get to /etc so it fails right away. I've modified that to fail a bit more gracefully, show the user a more human friendly message, and still produce the archive.

The other issue that's 6.x specific is that it appears that the nodes output no longer has a "conf" entry in the settings json in the nodes output. The only place it appears to be referenced is in the input arguments, so I've added a check to go through those checking for the configuration directory, since I need that location to find it on the node. 

I just tested on a plain vanilla install that should be similar to what you have and it's completing both from the elevated and normal prompts, although if you don't have sufficient permissions to read the log and/or config you won't see any artifacts bundled in the archive.
